lib/Sema/TypeCheckAttr.cpp

```diff
@@ -1831,6 +1831,18 @@ synthesizeMainBody(AbstractFunctionDecl *fn, void *arg) {
   Expr *returnedExpr;
 
   if (mainFunction->hasAsync()) {
+    if (!context.LangOpts.EnableExperimentalConcurrency) {
+      context.Diags.diagnose(mainFunction->getAsyncLoc(),
```

Comment:
We may want a comment relating how this fixes the crash because `_Concurrency` implicitly gets imported when `-enable-experimental-concurrency` is passed in, but doesn't when it's not. If that changes, the crash comes back.

Reply:
It'll work the way it is now, but how about checking for `!concurrencyModule` below instead of `-enable-experimental-concurrency`? That would avoid the crash if the module doesn't get loaded for any reason.

Reply:
Okay, now checks against the concurrency module. Also added a test passing both `-enable-experimental-concurrency` and `-parse-stdlib` since that won't implicitly import `_Concurrency` either.